core/src/main/java/io/parsingdata/metal/expression/comparison/ComparisonExpression.java

```diff
@@ -57,7 +59,7 @@ public ComparisonExpression(final ValueExpression value, final ValueExpression p
 
     @Override
     public boolean eval(final ParseGraph graph, final Encoding encoding) {
-        final ImmutableList<Optional<Value>> values = value == null ? ImmutableList.create(Optional.of(graph.current())) : value.eval(graph, encoding);
+        final ImmutableList<Optional<Value>> values = value == null ? ImmutableList.create(graph.current().map(identity())) : value.eval(graph, encoding);
```

Comment:
The null check for value makes me think: wouldn't it be nice if we can remove all the null values in the shorthands? I think we could have an `Empty` ValueExpression (or something) and overload the constructor with only a predicate and initialize value with the Empty ValueExpression. We then let this empty value expression return `ImmutableList.create(graph.current().map(identity()))` when `value.eval(...)` and it is sufficient to simply call for `value.eval(graph, encoding)` here. This can be adjusted for all the null usages in the shorthands.

Just a thought! I'll stop typing now...

[Edit: Oh hee, there already is the `Self` ValueExpression which contains `ImmutableList.create(graph.current().map(identity()))` in its `eval` so we could use that? Ok, I really stop typing now!]

`<MUTE OFF>`
[Edit 2:  Oh! Or... simple lambda expressions! (I see different implementation for when value == null). Would be nice to have a nullless metal! ]
`<MUTE ON>`